File: backend/app/schemas/conversation.py
"""
Conversation schemas.
"""

from datetime import datetime

from pydantic import BaseModel
from pydantic import ConfigDict


class ConversationResponse(BaseModel):
    """
    Conversation API response.
    """

    model_config = ConfigDict(
        from_attributes=True
    )

    id: int

    organization_id: int

    agent_id: int | None

    user_id: int | None

    session_id: str

    status: str

    created_at: datetime

    updated_at: datetime


class ConversationStatusUpdate(BaseModel):
    """
    Update conversation status.
    """

    status: str


class MessageResponse(BaseModel):
    """
    Conversation message response.
    """

    model_config = ConfigDict(
        from_attributes=True
    )

    id: int

    conversation_id: int

    role: str

    content: str

    created_at: datetime

File: frontend/src/types/conversation.ts
export interface Conversation {
  id: number;

  organization_id: number;

  user_id: number | null;

  session_id: string;

  status: string;

  created_at: string;

  updated_at: string;
}

export interface Message {
  id: number;

  conversation_id: number;

  role: string;

  content: string;

  created_at: string;
}